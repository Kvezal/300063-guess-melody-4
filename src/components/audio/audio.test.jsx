import React from "react";
import render from "react-test-renderer";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Audio from "./audio";


Enzyme.configure({
  adapter: new Adapter(),
});

describe(`AudioComponent`, () => {
  test(`should render component`, () => {
    const tree = render.create(<Audio onPlayAudioClick={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should call onPlayAudioClick`, () => {
    const onPlayAudioClickMock = jest.fn();
    const audio = mount(<Audio onPlayAudioClick={onPlayAudioClickMock} />);
    const playButton = audio.find(`button`);
    playButton.simulate(`click`);
    expect(onPlayAudioClickMock).toHaveBeenCalled();
  });
});
