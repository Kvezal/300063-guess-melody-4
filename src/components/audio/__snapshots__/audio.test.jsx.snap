// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioComponent should render component 1`] = `
Array [
  <button
    className="play-button play-button--small "
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
